lib/hocon/config_resolve_options.rb

```diff
@@ -0,0 +1,29 @@
+require 'hocon'
+
+class Hocon::ConfigResolveOptions
+  attr_reader :use_system_environment, :allow_unresolved
+
+  def initialize(use_system_environment, allow_unresolved)
+    @use_system_environment = use_system_environment
+    @allow_unresolved = allow_unresolved
+  end
+
+  def set_use_system_environment(value)
+    self.new(value, @allow_unresolved)
+  end
+
+  def set_allow_resolved(value)
+    self.new(value, @allow_unresolved)
```

Comment:
This should be `set_allow_unresolved`. Also it looks like the value is being used to set `use_system_environment` rather than `allow_unresolved`.